src/statement-list/statement_list_typecheck.cpp

```diff
@@ -1208,11 +1326,17 @@ void statement_list_typecheckt::typecheck_CPROVER_assume(
     expr_try_dynamic_cast<equal_exprt>(op_code.op1());
   if(assignment)
   {
-    const symbol_exprt &rhs{to_symbol_expr(assignment->rhs())};
-    // Check if identifier is present, create assumption and add it to the
-    // function body.
-    const exprt op{typecheck_identifier(tia_element, rhs.get_identifier())};
-    const code_assumet assumption{op};
+    exprt assigned_operand;
```

Comment:
Exactly the same code as above, except for assumptions vs. assertions. Either extract to `static` function or handle them in the same branch, and only differentiate once you instantiate either `code_assertt` or `code_assumptiont`.